instrumentation/opentelemetry_req/test/opentelemetry_req_test.exs

```diff
@@ -17,48 +25,365 @@ defmodule OpentelemetryReqTest do
+    ]
+
+    for {attr, expected} <- expected_attrs do
+      actual = Map.get(attrs, attr)
+      assert expected == actual, "#{attr} expected #{expected} got #{inspect(actual)}"
+    end
+
+    user_agent = Map.get(attrs, UserAgentAttributes.user_agent_original())
+    assert String.starts_with?(user_agent, "req/")
+  end
+
+  describe "errors" do
+    test "timeout exception error" do
+      Req.get(client(),
+        plug: fn conn ->
+          Req.Test.transport_error(conn, :timeout)
+        end,
+        retry: false,
+        url: "/"
+      )
+
+      expected_status = OpenTelemetry.status(:error, "timeout")
+
+      assert_receive {:span, span(attributes: span_attrs, status: ^expected_status)}
+
+      attrs = :otel_attributes.map(span_attrs)
+
+      expected_attrs = [
+        {ErrorAttributes.error_type(), Req.TransportError}
+      ]
+
+      for {attr, expected} <- expected_attrs do
+        actual = Map.get(attrs, attr)
+        assert expected == actual, "#{attr} expected #{expected} got #{inspect(actual)}"
+      end
+    end
+
+    test "4xx level error", %{bypass: bypass} do
+      Bypass.expect_once(bypass, "GET", "/", fn conn ->
+        conn
+        |> Plug.Conn.put_status(404)
+        |> Req.Test.text("not found")
+      end)
+
+      Req.get(client(),
+        retry: false,
+        url: "http://localhost:#{bypass.port}"
+      )
+
+      expected_status = OpenTelemetry.status(:error, "")
+
+      assert_receive {:span, span(attributes: span_attrs, status: ^expected_status)}
+
+      attrs = :otel_attributes.map(span_attrs)
+
+      expected_attrs = [
+        {ErrorAttributes.error_type(), "404"}
+      ]
+
+      for {attr, expected} <- expected_attrs do
+        actual = Map.get(attrs, attr)
+        assert expected == actual, "#{attr} expected #{expected} got #{inspect(actual)}"
+      end
+    end
+
+    test "5xx level error", %{bypass: bypass} do
+      Bypass.expect_once(bypass, "GET", "/", fn conn ->
+        conn
+        |> Plug.Conn.put_status(500)
+        |> Req.Test.text("internal server error")
+      end)
+
+      Req.get(client(),
+        retry: false,
+        url: "http://localhost:#{bypass.port}"
+      )
+
+      expected_status = OpenTelemetry.status(:error, "")
+
+      assert_receive {:span, span(attributes: span_attrs, status: ^expected_status)}
+
+      attrs = :otel_attributes.map(span_attrs)
+
+      expected_attrs = [
+        {ErrorAttributes.error_type(), "500"}
+      ]
+
+      for {attr, expected} <- expected_attrs do
+        actual = Map.get(attrs, attr)
+        assert expected == actual, "#{attr} expected #{expected} got #{inspect(actual)}"
+      end
+    end
+  end
+
+  # exception test
+  def ok_resp(conn) do
+    conn |> Req.Test.text("ok")
+  end
+
+  describe "span name" do
+    test "with no path params" do
+      Req.get!(client(),
+        plug: &ok_resp/1,
+        url: "/"
+      )
+
+      assert_receive {:span, span(name: :GET)}
```

Comment:
Is it weird that the name is an atom here and a binary when the path params are parsed? Feels surprising to me.

Reply:
We don't `to_atom` anything for users as a general rule. I'm not sure I'd want to deviate from it since sobelow would probably gripe but I don't know that for a fact. What do you think?

Reply:
Right, I don't think we'd convert to an atom, but I was kind of expecting that the span name would always be a string.

Reply:
Similar to the other comment, I don't think this is a blocker if that's already how it was working before, but I agree that it would be something worth considering as an option in a future PR where we can focus on just that aspect.